### PR TITLE
Add @CacheKey along with @iterableCacheKey

### PR DESCRIPTION
**Pull request by jakubtiteo**

### CODE REVIEW

File: packages/cached/lib/src/templates/cached_method_template.dart [hunk 1]
```diff
@@ -114,14 +115,47 @@ if (currentTtl != null && currentTtl.isBefore(now)) {
...
+    );
+  }
+
+  String _generateParamKeyPartCall({
+    required String name,
+    required CacheKeyAnnotation? cacheKeyAnnotation,
+    required bool oneParamKey,
+  }) =>
+      cacheKeyAnnotation != null
+          ? oneParamKey
```
Comment (Axot017) [inline packages/cached/lib/src/templates/cached_method_template.dart:147]: I think we don't need this 'oneParam' logic. It's making generator code much harder to read, and generated code is doing exactly the same thing.

File: packages/cached_annotation/lib/src/cache_key.dart [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  if(l == null) {
+    return l.hashCode.toString();
+  }
+
+  const prime = 31;
+  var result = 1;
+  for (final i in l as Iterable) {
+    result = result * prime + i.hashCode;
+  }
+  return result.toString();
```
Comment (Axot017) [inline packages/cached_annotation/lib/src/cache_key.dart:51]:
```suggestion
  return result.toString();
```

Comment (Axot017) [inline packages/cached_annotation/lib/src/cache_key.dart:51]:
```suggestion
  if(l == null || l is! Iterable) {
    return l.hashCode.toString();
  }
  
  return Object.hashAll(l);
```

**Review by Axot017 — APPROVED** [review]
lgtm

**Review by RSC-88 — APPROVED** [review]
lpg

**Pull request merged**